### PR TITLE
Make extractors work with resource packs

### PR DESCRIPTION
**Pull request by kalenedrael**
Currently, it's not possible to add a new texture for extractors. This registers the custom extractor models with Forge so it knows to stitch the corresponding texture into the atlas.

It's also not possible to change the extractor model, because the models are baked on first use and never updated. This adds event handlers to detect a resource pack reload and invalidate the cached models.

Also fix a couple of build issues:
- change modmaven.k-4u.nl to modmaven.dev (modmaven.k-4u.nl redirects to modmaven.dev, but modmaven.k-4u.nl has an expired certificate)
- make gradlew executable

Also add a convenience function to generate ResourceLocation.

### CODE REVIEW

File: src/main/java/de/maxhenkel/pipez/events/ModelEvents.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import de.maxhenkel.pipez.Main;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.client.event.ModelRegistryEvent;
+import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+@Mod.EventBusSubscriber(modid = Main.MODID, bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
```
Comment (henkelmax) [inline src/main/java/de/maxhenkel/pipez/events/ModelEvents.java:10]: Please register the event in the Main class for consistency.

File: src/main/java/de/maxhenkel/pipez/Main.java [hunk 2]
```diff
@@ -80,4 +81,8 @@ public void clientSetup(FMLClientSetupEvent event) {
         Containers.clientSetup();
     }
 
+    public static ResourceLocation rl(String location) {
+        return new ResourceLocation(MODID, location);
+    }
+
```
Comment (henkelmax) [inline src/main/java/de/maxhenkel/pipez/Main.java:87]: Please leave the resource locations like they are. I do it like this in every of my mods. I want to keep it consistent with my other mods.

**Review by henkelmax — CHANGES_REQUESTED** [review]

File: src/main/java/de/maxhenkel/pipez/Main.java (hunk 2 above)
Comment (kalenedrael) [reply src/main/java/de/maxhenkel/pipez/Main.java]: Sure thing.

File: src/main/java/de/maxhenkel/pipez/events/ModelEvents.java (hunk 1 above)
Comment (kalenedrael) [reply src/main/java/de/maxhenkel/pipez/events/ModelEvents.java]: Done.

**Comment by henkelmax** [comment]
Please don't do commits that revert changes. Just remove the commit from the PR with a force push or something.

**Comment by kalenedrael** [comment]
Done.

**Comment by henkelmax** [comment]
You want to merge to master, but master is the branch for 1.17.1. Please change it to 1.16.5

**Comment by kalenedrael** [comment]
Done.

File: src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java [hunk 3]
```diff
@@ -36,6 +39,8 @@ public PipeRenderer(TileEntityRendererDispatcher rendererDispatcherIn) {
             IUnbakedModel modelOrMissing = ModelLoader.instance().getModelOrMissing(getModel());
             return modelOrMissing.bake(ModelLoader.instance(), ModelLoader.instance().getSpriteMap()::getSprite, ModelRotation.X0_Y0, getModel());
         });
+
+        FMLJavaModLoadingContext.get().getModEventBus().register(this);
```
Comment (henkelmax) [inline src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java:43]: Please only register events in Main

**Review by henkelmax — CHANGES_REQUESTED** [review]

File: src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java (hunk 3 above)
Comment (kalenedrael) [reply src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java]: Done.

File: src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java [hunk 4]
```diff
@@ -17,16 +17,16 @@
...
 import net.minecraftforge.client.model.data.EmptyModelData;
-import net.minecraftforge.eventbus.api.SubscribeEvent;
-import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public abstract class PipeRenderer extends TileEntityRenderer<PipeTileEntity> {
 
+    private static List<PipeRenderer> renderers = new ArrayList();
```
Comment (henkelmax) [inline src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java:28]: There has to be a cleaner method than storing all renderers in a list or registering a event for every renderer.

**Review by henkelmax — CHANGES_REQUESTED** [review]

File: src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java (hunk 4 above)
Comment (kalenedrael) [reply src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java:28]: Every renderer has a cached model, and all cached models have to be invalidated. How about registering all cached models in ModelEvents?

Comment (kalenedrael) [reply src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java]: This might be a better solution.

**Review by henkelmax — APPROVED** [review]

**Comment by henkelmax** [comment]
Thank you very much!

**Comment by kalenedrael** [comment]
Thanks for merging. I created this texture pack to work with Pipez: https://www.curseforge.com/minecraft/texture-packs/pipez-retexture

**Comment by henkelmax** [comment]
> Thanks for merging. I created this texture pack to work with Pipez: https://www.curseforge.com/minecraft/texture-packs/pipez-retexture

Looks nice!

**Pull request merged**